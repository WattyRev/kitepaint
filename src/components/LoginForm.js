import React from "react";
import PropTypes from "prop-types";
import { A, P, Label, Input, Button } from "../theme";

/**
 * A from for logging in to KitePaint.
 */
const LogInForm = ({
  id,
  isDisabled,
  onPasswordChange,
  onRegister,
  onResetPassword,
  onSubmit,
  onUsernameChange,
  password,
  username
}) => (
  <form
    id={id}
    onSubmit={e => {
      e.preventDefault();
      onSubmit();
    }}
  >
    <Label htmlFor={`${id}-username`}>Username</Label>
    <Input
      id={`${id}-username`}
      value={username}
      disabled={isDisabled}
      onChange={e => onUsernameChange(e.target.value)}
    />
    <Label htmlFor={`${id}-password`}>Password</Label>
    <Input
      id={`${id}-password`}
      type="password"
      value={password}
      disabled={isDisabled}
      onChange={e => onPasswordChange(e.target.value)}
    />
    <Button isPrimary isBlock type="submit" disabled={isDisabled}>
      Sign In
    </Button>
    <P>
      <A onClick={onResetPassword}>Lost your password?</A>
    </P>
    <P>
      <A onClick={onRegister}>Register</A>
    </P>
  </form>
);

LogInForm.propTypes = {
  /**
   * An identifieng string used to differentiate this from other instances
   */
  id: PropTypes.string.isRequired,
  /**
   * Indicates if the form should be disabled.
   */
  isDisabled: PropTypes.bool,
  /**
   * A function called when the password field changes. Is called with the new value as the first
   * parameter
   */
  onPasswordChange: PropTypes.func.isRequired,
  /**
   * A function called when the link to the register form is clicked.
   */
  onRegister: PropTypes.func.isRequired,
  /**
   * A function called when the link to the reset password form is clicked.
   */
  onResetPassword: PropTypes.func.isRequired,
  /**
   * A function called when the form is submitted.
   */
  onSubmit: PropTypes.func.isRequired,
  /**
   * A function called when the username is changed. Is called with the new username as the first
   * parameter.
   */
  onUsernameChange: PropTypes.func.isRequired,
  /**
   * The value for the password field.
   */
  password: PropTypes.string,
  /**
   * The value for the username field.
   */
  username: PropTypes.string
};

export default LogInForm;
