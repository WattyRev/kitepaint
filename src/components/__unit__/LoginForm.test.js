import React from "react";
import { shallow } from "enzyme";
import LoginForm from "../LoginForm";

describe("LogInForm", () => {
  let defaultProps;
  beforeEach(() => {
    defaultProps = {
      id: "abc",
      onPasswordChange: jest.fn(),
      onRegister: jest.fn(),
      onResetPassword: jest.fn(),
      onSubmit: jest.fn(),
      onUsernameChange: jest.fn()
    };
  });
  it("renders", () => {
    expect.assertions(1);
    const wrapper = shallow(<LoginForm {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
