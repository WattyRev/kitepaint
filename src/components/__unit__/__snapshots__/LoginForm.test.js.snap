// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInForm renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogInForm
    id="abc"
    onPasswordChange={[MockFunction]}
    onRegister={[MockFunction]}
    onResetPassword={[MockFunction]}
    onSubmit={[MockFunction]}
    onUsernameChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef
          htmlFor="abc-username"
        >
          Username
        </ForwardRef>,
        <Input
          id="abc-username"
          onChange={[Function]}
        />,
        <ForwardRef
          htmlFor="abc-password"
        >
          Password
        </ForwardRef>,
        <Input
          id="abc-password"
          onChange={[Function]}
          type="password"
        />,
        <Button
          isBlock={true}
          isPrimary={true}
          type="submit"
        >
          Sign In
        </Button>,
        <P>
          <ForwardRef
            onClick={[MockFunction]}
          >
            Lost your password?
          </ForwardRef>
        </P>,
        <P>
          <ForwardRef
            onClick={[MockFunction]}
          >
            Register
          </ForwardRef>
        </P>,
      ],
      "id": "abc",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Username",
          "htmlFor": "abc-username",
        },
        "ref": null,
        "rendered": "Username",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "rules": Array [
              "
  margin: 8px 0 4px;
  display: block;
",
            ],
          },
          "displayName": "Styled(Label)",
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": undefined,
          "id": "abc-username",
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Password",
          "htmlFor": "abc-password",
        },
        "ref": null,
        "rendered": "Password",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "rules": Array [
              "
  margin: 8px 0 4px;
  display: block;
",
            ],
          },
          "displayName": "Styled(Label)",
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": undefined,
          "id": "abc-password",
          "onChange": [Function],
          "type": "password",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Sign In",
          "disabled": undefined,
          "isBlock": true,
          "isPrimary": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Sign In",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            onClick={[MockFunction]}
          >
            Lost your password?
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Lost your password?",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Lost your password?",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  text-decoration: none;
  margin: 0;
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
",
              ],
            },
            "displayName": "Styled(A)",
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            onClick={[MockFunction]}
          >
            Register
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Register",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Register",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  text-decoration: none;
  margin: 0;
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
",
              ],
            },
            "displayName": "Styled(A)",
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef
            htmlFor="abc-username"
          >
            Username
          </ForwardRef>,
          <Input
            id="abc-username"
            onChange={[Function]}
          />,
          <ForwardRef
            htmlFor="abc-password"
          >
            Password
          </ForwardRef>,
          <Input
            id="abc-password"
            onChange={[Function]}
            type="password"
          />,
          <Button
            isBlock={true}
            isPrimary={true}
            type="submit"
          >
            Sign In
          </Button>,
          <P>
            <ForwardRef
              onClick={[MockFunction]}
            >
              Lost your password?
            </ForwardRef>
          </P>,
          <P>
            <ForwardRef
              onClick={[MockFunction]}
            >
              Register
            </ForwardRef>
          </P>,
        ],
        "id": "abc",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Username",
            "htmlFor": "abc-username",
          },
          "ref": null,
          "rendered": "Username",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": true,
              "rules": Array [
                "
  margin: 8px 0 4px;
  display: block;
",
              ],
            },
            "displayName": "Styled(Label)",
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": undefined,
            "id": "abc-username",
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Password",
            "htmlFor": "abc-password",
          },
          "ref": null,
          "rendered": "Password",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": true,
              "rules": Array [
                "
  margin: 8px 0 4px;
  display: block;
",
              ],
            },
            "displayName": "Styled(Label)",
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": undefined,
            "id": "abc-password",
            "onChange": [Function],
            "type": "password",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Sign In",
            "disabled": undefined,
            "isBlock": true,
            "isPrimary": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              onClick={[MockFunction]}
            >
              Lost your password?
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Lost your password?",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Lost your password?",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "rules": Array [
                  "
  color: ",
                  [Function],
                  ";
  text-decoration: none;
  margin: 0;
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
",
                ],
              },
              "displayName": "Styled(A)",
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              onClick={[MockFunction]}
            >
              Register
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Register",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Register",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "rules": Array [
                  "
  color: ",
                  [Function],
                  ";
  text-decoration: none;
  margin: 0;
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
",
                ],
              },
              "displayName": "Styled(A)",
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
